### PR TITLE
Don't abort if weakremover.inc don't exist on source project

### PR DESCRIPTION
**Pull request by fcrozat**
For ALP, weakremover.inc might not be part of the staging parent project. Just ignore 404 error in that case.

### CODE REVIEW

**Review by Vogtinator — CHANGES_REQUESTED** [review]
IMO this needs explicit error handling for all possible cases:

(exists in staging, doesn't exist in staging) * (exists in target project, doesn't exist in target project)

In the case that there's a mismatch I'm not sure what it should do though. IMO abort and ask the user to resolve it manually.

Currently it silently ignores any possible errors which is bad.

**Review by Vogtinator — APPROVED** [review]
Should ideally check for 404 but it didn't do that before either :shrug:

File: osclib/freeze_command.py [hunk 1]
```diff
@@ -243,11 +243,20 @@ def copy_weakremovers(self):
...
+            targeturl_exists=True
         except HTTPError:
-            # if it doesn't exist, don't update
+            targeturl_exists=False
+        try:
+            sourceurl = self.api.makeurl(['source', self.api.project, '000release-packages', 'weakremovers.inc'])
+            inc = osc.core.http_GET(sourceurl).read()
+            sourceurl_exists=True
+        except HTTPError:
+            sourceurlurl_exists=False
```
Comment (Vogtinator) [inline osclib/freeze_command.py:254]:
```suggestion
            sourceurl_exists=False
```

**Review by Vogtinator — APPROVED** [review]

**Comment by Vogtinator** [comment]
I'd merge, but now there are unrelated commits on top

**Comment by fcrozat** [comment]
> I'd merge, but now there are unrelated commits on top

let me fix that, I should have used a topic branch in the first place..

**Comment by fcrozat** [comment]
switching to https://github.com/openSUSE/openSUSE-release-tools/pull/3083

**Pull request closed without merging**